plugins/org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/FileLocationsImpl.xtend

```diff
@@ -14,20 +14,20 @@ import org.eclipse.xtend.lib.macro.file.FileLocations
 import org.eclipse.xtend.lib.macro.file.Path
 import org.eclipse.xtext.generator.IContextualOutputConfigurationProvider
 import org.eclipse.xtext.workspace.IProjectConfig
-import org.eclipse.xtext.workspace.IWorkspaceConfigProvider
+import org.eclipse.xtext.workspace.IProjectConfigProvider
 
 /**
  * @author Sven Efftinge - Initial contribution and API
  */
 class FileLocationsImpl implements FileLocations {
 	
-	@Inject @Accessors IWorkspaceConfigProvider projectInformationProvider
+	@Inject @Accessors IProjectConfigProvider projectInformationProvider
 	@Inject @Accessors IContextualOutputConfigurationProvider outputConfigurationProvider
 	@Accessors Resource context
 	
 	protected def IProjectConfig getProjectConfig(Path path) {
 		val string = path.getSegments().get(0)
-		val projectConfig = projectInformationProvider.getWorkspaceConfig(context.resourceSet).findProjectByName(string)
+		val projectConfig = projectInformationProvider.getProjectConfig(context.resourceSet)
```

Comment:
We will now return the same project independently from the used path. We should at least assert that the given path matches the returned project.